File: ClassLoaderHelper/src/main/java/tests/ClassToLoadMultipleTimes.java
package tests;

import java.net.URL;

public class ClassToLoadMultipleTimes
{
	static
	{
		System.out.println( "ClassToLoadMultipleTimes" );
	}
	
	static String getClassPath( Class<?> clazz )
	  {
	    ClassLoader loader = clazz.getClassLoader();
	    if ( loader == null )
	      return null;
	    URL url = loader.getResource( clazz.getName().replace('.', '/' )
	                                   + ".class" );
	    return ( url != null ) ? url.toString() : null;
	  }
	
	  public static void main( String... args ) throws Exception
	  {
		Class<?> c = Class.forName( "tests.ClassToLoadMultipleTimes" );
	    System.out.println( "Klasse: " + c.getName() );
	    System.out.println( "Dateiname: " + getClassPath(c) );
	  }
}

File: ClassLoaderHelper/src/main/java/tests/ClassToLoadMultipleTimes.kt
package tests

import java.net.URL

@Suppress("unused") // used by reflection
class ClassToLoadMultipleTimes {
    init {
        println("ClassToLoadMultipleTimes")
    }
}

internal fun getClassPath(clazz: Class<*>): URL? =
        clazz.classLoader.getResource(clazz.name.replace('.', '/') + ".class")

fun main() {
    val c = Class.forName("tests.ClassToLoadMultipleTimes")
    println("Klasse: ${c.name}")
    println("Dateiname: ${getClassPath(c)}")
}
